src/components/CustomerQuota/NoQuota/NoQuotaCard.tsx

```diff
@@ -0,0 +1,210 @@
+import { TransactionsByCategory, Transaction } from "./TransactionsByCategory";
+import { ShowFullListToggle } from "./ShowFullListToggle";
+import {
+  DistantTransactionTitle,
+  RecentTransactionTitle,
+  NoPreviousTransactionTitle
+} from "./TransactionTitle";
+import { AppealButton } from "./AppealButton";
+
+const DURATION_THRESHOLD_SECONDS = 60 * 10; // 10 minutes
+const MAX_TRANSACTIONS_TO_DISPLAY = 5;
+
+interface NoQuotaCard {
+  ids: string[];
+  cart: Cart;
+  onCancel: () => void;
+  onAppeal?: () => void;
+}
+/**
+ * Shows when the user cannot purchase anything
+ *
+ * Precondition: Only rendered when all items in cart have max quantity of 0
+ */
+export const NoQuotaCard: FunctionComponent<NoQuotaCard> = ({
+  ids,
+  cart,
+  onCancel,
+  onAppeal
+}) => {
+  const [isShowFullList, setIsShowFullList] = useState<boolean>(false);
+  const { getProduct, allProducts } = useProductContext();
+  const { token, endpoint } = useAuthenticationContext();
+
+  const policyType = cart.length > 0 && getProduct(cart[0].category)?.type;
+
+  const { pastTransactionsResult } = usePastTransaction(ids, token, endpoint);
+  // Assumes results are already sorted (valid assumption for results from /transactions/history)
+  const sortedTransactions = pastTransactionsResult?.pastTransactions;
+
+  const latestTransactionTime: Date | undefined =
+    sortedTransactions?.[0].transactionTime ?? undefined;
+  const now = new Date();
+  const secondsFromLatestTransaction = latestTransactionTime
+    ? differenceInSeconds(now, latestTransactionTime)
+    : -1;
+
+  const hasAppealProduct = (): boolean => {
+    return allProducts.some(policy => policy.categoryType === "APPEAL");
+  };
+
+  // Group transactions by category
+  const transactionsByCategoryMap: {
+    [category: string]: {
+      transactions: Transaction[];
+      hasLatestTransaction?: boolean;
+    };
+  } = {};
+  sortedTransactions?.forEach(item => {
+    const policy = allProducts.find(
+      policy => policy.category === item.category
+    );
+    const categoryName = policy?.name ?? item.category;
+    const formattedDate = format(item.transactionTime, "d MMM yyyy, h:mma");
+
+    if (!(categoryName in transactionsByCategoryMap)) {
+      transactionsByCategoryMap[categoryName] = {
+        transactions: [],
+        hasLatestTransaction: false
+      };
+    }
+    transactionsByCategoryMap[categoryName].transactions.push({
+      transactionDate: formattedDate,
+      details: getAllIdentifierInputDisplay(item.identifierInputs ?? []),
+      quantity: formatQuantityText(
+        item.quantity,
+        policy?.quantity.unit || { type: "POSTFIX", label: " qty" }
+      ),
+      isAppeal: policy?.categoryType === "APPEAL"
+    });
+    transactionsByCategoryMap[categoryName].hasLatestTransaction =
+      transactionsByCategoryMap[categoryName].hasLatestTransaction ||
+      differenceInSeconds(latestTransactionTime || 0, item.transactionTime) ===
+        0;
+  });
```

Comment:
Hmmm ideally should be writing test for the functional component 🙈 Not sure about benefit of refactoring this because it is only used by this functional component. What do you think if I moved this to a method outside of the component and wrote a test for it? I'm thinking to keep the util function in the same file since only this file uses it

Reply:
Yep, agreed, lets move it outside the component, then write tests for it. Alternatively can write integration tests for this noQuota screen 😊

Reply:
Done!! Broke them up in a way that I think makes sense, do let me know if you think can be broken down more :)